volmdlr/faces.py

```diff
@@ -5118,455 +5159,381 @@ def is_adjacent(self, face2: 'Face3D'):
             return True
         return False
 
+    def edge3d_inside(self, edge3d):
+        method_name = f'{edge3d.__class__.__name__.lower()[:-2]}_inside'
+        if hasattr(self, method_name):
+            return getattr(self, method_name)(edge3d)
+        points = edge3d.discretization_points(number_points=5)
+        for point in points[1:-1]:
+            if not self.point_belongs(point):
+                return False
+        return True
```

Comment:
displaced from PlaneFace3D and generalised to work with any edge. In short it verifies if an edge lies completely on the face3D.